### PR TITLE
Feature/cleanup directory build props and workflow

### PR DESCRIPTION
**Pull request by perkops**

### CODE REVIEW

**Review by davidkallesen — APPROVED** [review]

File: global.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "sdk": {
+      "rollForward": "latestMajor",
+      "allowPrerelease": false
+  }
+}
```
Comment (LarsSkovslund) [inline global.json:6]: Why do you feel it is required to introduce `global.json`, what issue does it address ?

File: src/Directory.Build.props [hunk 2]
```diff
@@ -28,30 +17,27 @@
...
+      Instruct the build system to embed project source files that are not tracked by the source control
       or imported from a source package to the generated PDB.
     -->
     <EmbedUntrackedSources>true</EmbedUntrackedSources>
     <!-- Recommended: Embed symbols containing Source Link in the main file (exe/dll) -->
     <DebugType>embedded</DebugType>
-
     <!-- Will generate nuget packages for each project -->
-    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
+    <GeneratePackageOnBuild>false</GeneratePackageOnBuild>
```
Comment (LarsSkovslund) [inline src/Directory.Build.props:38]: Why are you disabling my option to have nuget package build on compile locally?

File: global.json (hunk 1 above)
Comment (davidkallesen) [reply global.json:6]: It is a wish from Egil, so you can have something VS in prerelease

File: src/Directory.Build.props (hunk 2 above)
Comment (davidkallesen) [reply src/Directory.Build.props:38]: Right no, we have a seperate workflow-step for dotnet pack

File: global.json (hunk 1 above)
Comment (LarsSkovslund) [reply global.json:6]: YAGNI - This should only be there if we have an issues that can't be address by other means.
However - looking at other open-source projects they do tend to have a `global.json` in the root folder, so lets keep it.

File: src/Directory.Build.props (hunk 2 above)
Comment (LarsSkovslund) [reply src/Directory.Build.props:38]: Yes I do know, the reason why this is set to `true` is for me to have a nuget package generated as part of my `local` build so I would not have to run `dotnet pack` from command-line.

> NOTE: for our template this should be disabled by default.

Comment (davidkallesen) [reply src/Directory.Build.props:38]: Then I think the right solution should be `false` and then you want to make a local package and want to take a shortcut - then you can make a temporarily `true`.

**Pull request merged**